src/hdmf_zarr/backend.py

```diff
@@ -1047,6 +1070,8 @@ def write_dataset(self, **kwargs):  # noqa: C901
             refs = list()
             type_str = list()
             for i, dts in enumerate(options['dtype']):
+                # TODO: This appears to be code to distinguish between region and object References
+                # and will be refactored/removed.
```

Comment:
```suggestion
```

Reply:
The referenced RegionReference code was removed in https://github.com/hdmf-dev/hdmf-zarr/pull/236